lib/galaxy/managers/interactivetool.py

```diff
@@ -18,125 +26,100 @@
 )
 from galaxy.model.base import transaction
 from galaxy.security.idencoding import IdAsLowercaseAlphanumEncodingHelper
-from galaxy.util.filelock import FileLock
 
 log = logging.getLogger(__name__)
 
-DATABASE_TABLE_NAME = "gxitproxy"
+gxitproxy = Table(
+    "gxitproxy",
+    MetaData(),
+    Column("key", String(16), primary_key=True),
+    Column("key_type", Text(), primary_key=True),
+    Column("token", String(32)),
+    Column("host", Text()),
+    Column("port", Integer()),
+    Column("info", Text()),
+)
 
 
-class InteractiveToolSqlite:
-    def __init__(self, sqlite_filename, encode_id):
-        self.sqlite_filename = sqlite_filename
-        self.encode_id = encode_id
+class InteractiveToolPropagatorSQLAlchemy:
+    """
+    Propagator for InteractiveToolManager implemented using SQLAlchemy.
+    """
+
+    def __init__(self, database_url, encode_id):
+        """
+        Constructor that sets up the propagator using a SQLAlchemy database URL.
+
+        :param database_url: SQLAlchemy database URL, read more on the SQLAlchemy documentation
+                             https://docs.sqlalchemy.org/en/20/core/engines.html#database-urls.
+        :param encode_id: A helper class that can encode ids as lowercase alphanumeric strings and vice versa.
+        """
+        self._engine = create_engine(database_url)
+        self._encode_id = encode_id
 
     def get(self, key, key_type):
-        with FileLock(self.sqlite_filename):
-            conn = sqlite3.connect(self.sqlite_filename)
-            try:
-                c = conn.cursor()
-                select = f"""SELECT token, host, port, info
-                            FROM {DATABASE_TABLE_NAME}
-                            WHERE key=? and key_type=?"""
-                c.execute(
-                    select,
-                    (
-                        key,
-                        key_type,
-                    ),
-                )
-                try:
-                    token, host, port, info = c.fetchone()
-                except TypeError:
-                    log.warning("get(): invalid key: %s key_type %s", key, key_type)
-                    return None
-                return dict(key=key, key_type=key_type, token=token, host=host, port=port, info=info)
-            finally:
-                conn.close()
+        with self._engine.connect() as conn:
+            stmt = select(*gxitproxy.c["token", "host", "port", "info"]).where(
+                gxitproxy.c.key == key,
+                gxitproxy.c.key_type == key_type,
+            )
+            cursor_result = conn.execute(stmt)
+            result = cursor_result.fetchone()
+
+            if result is None:
+                log.warning("get(): invalid key: %s key_type %s", key, key_type)
+                return None
+
+            return dict(key=key, key_type=key_type, **dict(zip(cursor_result.keys(), result)))
```

Comment:
If you don't select individual columns, i.e. 
`stmt = select(gxitproxy).where(gxitproxy.c.key == key, gxitproxy.c.key_type == key_type)`
then your second return statement can be simplified to 
`return dict(zip(cursor_result.keys(), result))`
Same result.

Reply:
Dead code removed, https://github.com/galaxyproject/galaxy/pull/18481#discussion_r1741962053